src/main/kotlin/com/zopa/ktor/opentracing/OpenTracingServer.kt

```diff
@@ -47,22 +53,24 @@ class OpenTracingServer(
             pipeline.insertPhaseAfter(ApplicationCallPipeline.Fallback, tracingPhaseFinish)
 
             pipeline.intercept(tracingPhaseStart) {
-                if (config.filters.any { it(call) }) return@intercept
+                if (feature.filters.any { it(call) }) return@intercept
 
                 val headers: MutableMap<String, String> = call.request.headers.toMap()
                 headers.remove("Authorization")
 
                 val clientSpanContext: SpanContext? = tracer.extract(Format.Builtin.HTTP_HEADERS, TextMapAdapter(headers))
                 if (clientSpanContext == null) log.info("Tracing context could not be found in request headers. Starting a new server trace.")
 
-                val pathUuid: PathUuid = context.request.path().UuidFromPath()
-                val spanName = "${context.request.httpMethod.value} ${pathUuid.path}"
+                val pathAndTags = context.request.path().toPathAndTags(feature.toBeTaggedAndReplaced)
```

Comment:
Here you can make the code a little more readable by destructuring the response:
```kotlin
val (path, tags) = context.request.path().toPathAndTags(feature.toBeTaggedAndReplaced)
```

Reply:
Oooh forgot you could do this in Kotlin, that's a lot nicer 👍